core/src/main/java/io/fabric8/maven/core/config/RuntimeMode.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.core.config;
+
+import com.google.common.base.Objects;
+
+import java.util.Properties;
+
+public enum RuntimeMode {
+
+    /**
+     * Create resources descriptors for vanilla Kubernetes
+     */
+    kubernetes(false, "Kubernetes"),
+
+    /**
+     * Use special OpenShift features like BuildConfigs
+     */
+    openshift(false, "OpenShift"),
+
+    /**
+     * Detect automatically whether running on OpenShift or Kuberentes.
+     * This is done by contacting an API server
+     */
+    auto(true, "Auto");
```

Comment:
Iiur then "PlatformMode" is used for creating the resources (i.e. for doing the two passes through the enricher chain), and "RuntimeMode" is for deployment (i.e. to select which client to use for applying resources).

What does then "auto" mean here ? I would also recommend to adapt the documentation to better reflect this distinction.